packages/ses/src/tame-symbol-constructor.js

```diff
@@ -50,14 +50,20 @@ export const tameSymbolConstructor = () => {
     },
   });
 
+  const sharedSymbolDescs = getOwnPropertyDescriptors(SharedSymbol);
   const originalDescsEntries = entries(
     getOwnPropertyDescriptors(OriginalSymbol),
   );
   const descs = fromEntries(
-    arrayMap(originalDescsEntries, ([name, desc]) => [
-      name,
-      { ...desc, configurable: true },
-    ]),
+    arrayMap(originalDescsEntries, ([name, desc]) => {
+      if (
+        sharedSymbolDescs[name] &&
+        sharedSymbolDescs[name].configurable === false
```

Comment:
nb: condition `name === 'prototype'` is also sufficient on Hermes

Reply:
@erights any ideas on how to best handle this non-compliance? It will likely bite us in other places.

Reply:
I've never run Hermes before. I'm on a Mac. What's the best way for me to reproduce this? Possible to reproduce under a debugger, as I saw in your screenshots?

Reply:
>  Short-hand methods are not supposed to have a .prototype property.

Current hypothesis. The correct fix is likely to make `SharedSymbol` a `function` function rather than a concise method. This probably would have been somewhat better all along anyway, since the original `Symbol` does, in most ways, act more like a `function` function than a concise method.

Reply:
But I'd like to repro locally and see what happens.

Reply:
> condition name === 'prototype' is also sufficient on Hermes

I like the configurability test better. Perhaps, for aesthetics, we should do both.

Reply:
> since the original `Symbol` does, in most ways, act more like a `function` function than a concise method.

It does throw on construction. We'd have to explicitly do `if (new.target) throw TypeError()`

Reply:
> But I'd like to repro locally and see what happens.

Pretty primitive, but you have a REPL and CLI. Download https://github.com/facebook/hermes/releases/tag/v0.12.0

```
$ ./hermes 
>> ({foo(){}}).foo.prototype
foo { [constructor]: [Function foo] }
```

Reply:
> I like the configurability test better. Perhaps, for aesthetics, we should do both.

I think neither is valid. We need to copy over the `.prototype` or the Symbol "constructor" just isn't correct (`getPrototypeOf(Object(Symbol())) !== Symbol.prototype`)

Reply:
> > since the original `Symbol` does, in most ways, act more like a `function` function than a concise method.
> 
> It does throw on construction. We'd have to explicitly do `if (new.target) throw TypeError()`

As a matter of fidelity, the spec does say `Symbol` has a `[[Construct]]` which throws on construction. It is possible to detect using the following:

```js
const hasConstruct = (target) => { 
  const proxy = new Proxy(target, {
    construct() { return {}; }
  });
  try {
    new proxy();
    return true;
  } catch (e) {
    return false;
  }
}
hasConstruct(Symbol); //true
hasConstruct(({Symbol() {}}).Symbol); // false
```

Reply:
I am not sure why the re-definition of `.prototype` fails since it is marked as `writable`. Anyway, we should switch to function as mentioned above.

I have also filed an issue upstream for the conformance of short hand methods: https://github.com/facebook/hermes/issues/1371

Reply:
thanks again for looking into this and raising the issue
and sorry i didn't spot Hermes fixed this [last week](https://github.com/facebook/hermes/commit/00f18c89c720e1c34592bb85a1a8d311e6e99599) in the [static_h](https://github.com/facebook/hermes/tree/static_h) branch
since Static Hermes it's few months from shipping, will proceed with changes
perhaps with a comment like this?
```js
// This fix can be removed when Hermes (containing the mistake)
// is deprecated for Static Hermes (containing the fix)
// which is a backwards compatible drop-in replacement
// so object literal short-hand methods no longer have have a prototype
```

Reply:
since the https://github.com/facebook/hermes/releases/tag/v0.12.0 CLI is shy of couple yrs old
i requested to help get an updated version (being discussed with team)

so i tested the commit fix by [building and running](https://github.com/facebook/hermes/blob/main/doc/BuildingAndRunning.md) Hermes, remembering to git checkout the [static_h](https://github.com/facebook/hermes/tree/static_h) branch after cloning / before building

Reply:
i'll get back to you @erights with repro's on RN 71+ and running them with Flipper and Hermes Debugger
debugging works well for me on RN 71 but not above (if the same for you i'll file an upstream issue)

Reply:
> will proceed with changes
> perhaps with a comment like this?

The changes I suggested are actually an improvement over what's currently there, and should be kept regardless of Hermes.

Reply:
updated ^ and tested working on RN 71 (containing Hermes bug) on Android
with SES shim built from current/suggested changes
(`dist/ses.cjs` generated from `yarn clean && yarn build`)

Reply:
> I've never run Hermes before. I'm on a Mac. What's the best way for me to reproduce this? Possible to reproduce under a debugger, as I saw in your screenshots?

> i'll get back to you @erights with repro's on RN 71+ and running them with Flipper and Hermes Debugger on Mac debugging works well for me on RN 71, but not above versions (if the same for you i'll file an upstream issue)

@erights here's the repro with @mhofman's fix applied in a [commit ](https://github.com/leotm/RN07117SES/commit/7c4c043266f85978f3a86c33249920b886d6571e)
https://github.com/leotm/RN07117SES

and i've added a ReadMe instructions to run it on Mac, debug with Flipper and repro the issues tracked in
- https://github.com/endojs/endo/issues/1891